### ISSUE TITLE
DESCRIPTION in project() needs newer cmake version

### ISSUE THREAD

**Issue by wieset** [rock-core/base-cmake#62]
Error on Ubuntu 16.04:

```
    CMake Error at CMakeLists.txt:2 (project):
      project with VERSION must use LANGUAGES before language names.
```

Reason is 2a4a3b9c09660bc0b6b8b7edbb4fc66e5eb5b0ac. `DESCRIPTION` in `project()` is only supported from cmake 3.9, but default cmake version for Ubuntu 16.04 is 3.5 and `cmake_minimum_required` is 3.0. Would you consider removing the `DESCRIPTION` to keep compatibility with 16.04?

**Comment by doudou** [comment]
> cmake_minimum_required is 3.0.

From my perspective, this is the bug.

> Would you consider removing the DESCRIPTION to keep compatibility with 16.04?

I would consider doing it for `base/cmake` given its rather central place. But, now, the CMake lib template has also migrated to using DESCRIPTION and I think it's time to move forward beyond 16.04. @2maz, opinion ?

You could also pin `base/cmake` to 63f49ee9cfc5db8ee87cc69f8f3cd3aa6cfc1e0f to avoid this commit. Generally speaking, you can use `autoproj log` to find out where you were before an update, and pin packages where they were before they broke.

**Comment by wieset** [comment]
Agreed, it's time to move to 18.04+. For now, if you opt to keep it compatible with 16.04 it would be nice, otherwise we pin the commit.

**Comment by 2maz** [comment]
I would also suggest to move forward.
